test/test-functions

```diff
@@ -28,6 +28,27 @@ STATEDIR="${BUILD_DIR:-.}/test/$(basename $(dirname $(realpath $0)))"
 STATEFILE="$STATEDIR/.testdir"
 TESTLOG="$STATEDIR/test.log"
 
+is_built_with_asan() {
+    if ! type -P objdump >/dev/null; then
+        ddebug "Failed to find objdump. Assuming systemd hasn't been built with ASAN."
+        return 1
+    fi
+
+    # Borrowed from https://github.com/google/oss-fuzz/blob/cd9acd02f9d3f6e80011cc1e9549be526ce5f270/infra/base-images/base-runner/bad_build_check#L182
+    local _asan_calls=$(objdump -dC $BUILD_DIR/systemd | egrep "callq\s+[0-9a-f]+\s+<__asan" -c)
+    if (( $_asan_calls < 1000 )); then
+        return 1
+    else
+        return 0
+    fi
```

Comment:
Hmm, I don't understand how this is supposed to work. Here:
```console
$ BUILD_DIR=build-asan; objdump -dC $BUILD_DIR/systemd | egrep "callq\s+[0-9a-f]+\s+<__asan" -c
22910
$ BUILD_DIR=build; objdump -dC $BUILD_DIR/systemd | egrep "callq\s+[0-9a-f]+\s+<__asan" -c     
0
```
The result is reverse.

Reply:
What about just doing this:
```sh
meson configure $BUILD_DIR | grep -P 'b_sanitize\s+address' >/dev/null
```
?

Reply:
The result isn't reverse because `bash` considers `0` to be true and `1` to be false. I agree that it's counterintuitive, but, unfortunately, it's how `bash` works.

I don't think I can't get `meson configure $BUILD_DIR` working on Ubuntu CI where [`BUILD_DIR` is set to `.`](https://salsa.debian.org/systemd-team/systemd/blob/master/debian/tests/upstream#L12). In fact, according to the output of `objdump`, even `systemd` isn't in `$BUILD_DIR`.

Reply:
Apart from Ubuntu CI, it's possible to build `systemd` with `ASan` without using `-Db_sanitize`, so I wouldn't rely on the option being set.

Reply:
@keszybz as another option, I considered using `return $(($_asan_calls < 1000))` instead of the whole `if` statement, but in the end decided that it was a bit too much `bash` magic. If you think, that one line is better, I could make use of it, though I'd prefer to leave everything as is.

Reply:
This also works:

```shell
is_built_with_asan() {
    ...
    (( $_asan_calls < 1000 )) || return 1
}
```

The `return 0` will be implicit.

Also, inside `(( ... ))` you don't need a `$` on variables, so `(( _asan_calls < 1000 ))` also works, though I think the `$` actually helps here since then variables look the same everywhere...

I think what you have right now is just fine.